pkg/logentry/stages/multiline.go

```diff
@@ -0,0 +1,167 @@
+package stages
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	ErrEmptyMultilineStageConfig = "multiline stage config cannot be empty"
+	ErrMultilineNameRequired     = "multiline stage pipeline name can be omitted but cannot be an empty string"
+	ErrFirstLineRequired         = "firstLine regular expression required"
+)
+
+// MultilineConfig contains the configuration for a multilineStage
+type MultilineConfig struct {
+	PipelineName    *string `mapstructure:"pipeline_name"`
+	FirstLineRegexp string  `mapstructure:"firstline"`
+	MaxWait         string  `mapstructure:"max_wait_time"`
```

Comment:
I think we should have a combination of max_wait_time and max_line. We can have this a second PR thought.